swap/swap.go

```diff
@@ -467,56 +311,59 @@ func (s *Swap) Balances() (map[enode.ID]int64, error) {
 	return balances, nil
 }
 
-// loadLastSentCheque loads the last cheque for a peer from the state store (persisted)
-// To be called with mutex already held
-// Caller must be careful that the same cheque isn't concurrently read and written by multiple routines
-func (s *Swap) loadLastSentCheque(peer enode.ID) (err error) {
-	//only load if the current instance doesn't already have this peer's
-	//last cheque in memory
+// loadLastReceivedCheque loads the last received cheque for the peer from the store
+func (s *Swap) loadLastReceivedCheque(p enode.ID) (*Cheque, error) {
+	s.storeLock.Lock()
```

Comment:
Let's verify with @janos that we actually need this lock and why.

My intuition tells me that LevelDB can handle concurrent accesses for different keys. But we may need to indeed lock it for concurrent accesses from the same Peer. In that case thought it might be sufficient to lock `receivedChequeKey` for the given peer.